aderyn_core/src/context/browser/peek.rs

```diff
@@ -0,0 +1,12 @@
+use crate::context::workspace_context::{ASTNode, WorkspaceContext};
+
+pub trait Peek {
+    /// Peek into the source code of a node
+    fn peek(&self, context: &WorkspaceContext) -> Option<String>;
+}
+
+impl Peek for ASTNode {
+    fn peek(&self, context: &WorkspaceContext) -> Option<String> {
+        context.get_source_code_of_node(self.id()?)
+    }
+}
```

Comment:
Right now, it only works on ASTNode but if my other PR https://github.com/Cyfrin/aderyn/pull/253 is merged then I can easily extend this to all nodes natively

Reply:
> Right now, it only works on ASTNode but if my other PR #253 is merged then I can easily extend this to all nodes natively

DONE ✅  (now every `impl Node` has a `peek()`)